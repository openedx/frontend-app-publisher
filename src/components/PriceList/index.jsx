import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';

import RenderInputTextField from '../RenderInputTextField';
import FieldLabel from '../FieldLabel';

const PriceList = ({
  disabled,
  extraInput,
  priceLabels,
  required,
}) => (
  <>
    {Object.keys(priceLabels).map(seatType => (
      <Field
        key={seatType}
        name={`prices.${seatType}`}
        component={RenderInputTextField}
        extraInput={{
          min: 1.00,
          step: 0.01,
          max: 10000.00,
          ...extraInput,
        }}
        type="number"
        label={(
          <FieldLabel
            text={`${priceLabels[seatType]} Price (USD)`}
            required
          />
)}
        disabled={disabled}
        required={required}
      />
    ))}
  </>
);

PriceList.propTypes = {
  disabled: PropTypes.bool,
  extraInput: PropTypes.shape(),
  priceLabels: PropTypes.shape(),
  required: PropTypes.bool,
};

PriceList.defaultProps = {
  disabled: false,
  extraInput: {},
  priceLabels: {},
  required: false,
};

export default PriceList;
