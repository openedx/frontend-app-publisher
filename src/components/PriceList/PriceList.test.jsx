import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import PriceList from './index';

describe('PriceList', () => {
  it('renders without any price labels', () => {
    const component = shallow(<PriceList priceLabels={{}} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders with price labels', () => {
    const priceLabels = { a: 'A', b: 'B' };
    const component = shallow(<PriceList priceLabels={priceLabels} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
