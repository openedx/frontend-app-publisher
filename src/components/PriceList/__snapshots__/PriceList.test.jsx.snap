// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceList renders with price labels 1`] = `
<Fragment>
  <Field
    component={[Function]}
    disabled={false}
    extraInput={
      Object {
        "max": 10000,
        "min": 1,
        "step": 0.01,
      }
    }
    key="a"
    label={
      <FieldLabel
        className=""
        extraText=""
        helpText=""
        id={null}
        optional={false}
        required={true}
        text="A Price (USD)"
      />
    }
    name="prices.a"
    required={false}
    type="number"
  />
  <Field
    component={[Function]}
    disabled={false}
    extraInput={
      Object {
        "max": 10000,
        "min": 1,
        "step": 0.01,
      }
    }
    key="b"
    label={
      <FieldLabel
        className=""
        extraText=""
        helpText=""
        id={null}
        optional={false}
        required={true}
        text="B Price (USD)"
      />
    }
    name="prices.b"
    required={false}
    type="number"
  />
</Fragment>
`;

exports[`PriceList renders without any price labels 1`] = `<Fragment />`;
